src/js/components/List/doc.js

```diff
@@ -47,6 +47,10 @@ export const doc = List => {
 
   DocumentedList.propTypes = {
     ...genericProps,
+    action: PropTypes.func.description(
+      `The interactive element property for the user to be able to pass a 
+      component within list.`,
```

Comment:
I think we should also add a few words about the arguments that action is receiving and what they mean ` action={(item, index) => {...}}`, similarly to how you documented it for Menu children few PRs ago.

Reply:
okay yeah that sounds good what were your thought about changing name of `action` ?

Reply:
I think renaming the `action` is a valid concern, and I also think it should go on a separate PR since it will require backward compatibility handling and I wouldn't want it to block this current PR.

Reply:
good point! I will fix the definition to finish this PR